node/src/db_config/persistent_configuration.rs

```diff
@@ -419,21 +419,20 @@ impl PersistentConfiguration for PersistentConfigurationReal {
         match self.get("max_block_count") {
             Ok(max_block_count) => match decode_u64(max_block_count) {
                 Ok(mbc_opt) => Ok(mbc_opt),
-                Err(TypedConfigLayerError::BadNumberFormat(value)) if value.is_empty() => Ok(None),
                 Err(e) => Err(PersistentConfigError::from(e)),
             },
             Err(e) => Err(PersistentConfigError::from(e)),
         }
     }
 
     fn set_max_block_count(&mut self, value: Option<u64>) -> Result<(), PersistentConfigError> {
-        self.simple_set_method(
-            "max_block_count",
-            match encode_u64(value) {
-                Ok(Some(mbc)) => mbc,
-                _ => "".to_string(),
+        match encode_u64(value) {
+            Ok(mbc_opt) => match self.dao.set("max_block_count", mbc_opt) {
+                Ok(_) => Ok(()),
+                Err(e) => Err(PersistentConfigError::from(e)),
             },
-        )
+            Err(e) => Err(PersistentConfigError::from(e)),
+        }
```

Comment:
@bertllll I tried to apply all your suggestions except for this one. This is the way I was able to pass a `None`. I hope you find it acceptable. `self.dao.set()` can take an `Option<String>` and will store `None` as null. `PersistentConfiguration::simple_set_method()` would need to check if the `value` parameter is already an `Option<String>` and might be `None` and simply pass that to `self.dao.set()` If `value: T` is a `None ` value of `Option<String>`, then `value.to_string()` fails because `Option<String>` does not implement the required `Display` trait. If I'm doing something wrong here please let me know.

Reply:
I see. It seems that I was wrong. Sorry. It was me who wrote the "simple_set_method()" which was a naive attempt to standardize at least some of the methods because I noticed a repetitive pattern in there. Unfortunately, there are many methods that cannot be treated so simply. I know about the limitations but when I suggested you those few things I believed it was right to do. It was before I understood that the "simple_set_method()" could've been more flexible if I'd designed it with the generic argument inside an Option, then you could use it even for you optional value. In fact, I think one day in the future my patient with my past mistakes will reach the limits and will refine it (which might mean giving up on unnaturally much pushing people to use this little code "if their implementation is simple enough to use it". I don't like this kind of additional consideration that's hard to answer straightly and that has nothing to do with elegance in the end.)

I think you did a good job. The only detail I came to in my mind is that you could simplify the match statements and the PersistendConfigError::from() expressions by use of the question mark. It's kind of calling for it but I know that this can differ from person to another. It's more about a personal style than readability. So I'm okay with this.

Reply:
Can I really use the question mark for both match arms? Perhaps it's easier to see with the RustRover/IntelliJ hints:
![image](https://github.com/MASQ-Project/Node/assets/60554948/fb5218da-4bfa-4d36-b7ca-e3587c723a50)

The inner error match arm is converting `ConfigDaoError` and outer error match arm is converting `TypedConfigLayerError`. If it can be simplified I'm happy to do so.